### PR TITLE
[WIP] update help command

### PR DESCRIPTION
**Pull request by kakshay21**
Add iters, threads, and verbose command in help.
see https://github.com/vstakhov/rspamd/issues/2054

### CODE REVIEW

**Comment by kakshay21** [comment]
@vstakhov Need little direction in adding `learning-rate` and `penalty-weight`.
These could be in the form of arrays and in that case, I think it's better to take that from any config file?

File: src/rspamadm/rescore.c [hunk 1]
```diff
@@ -184,6 +196,12 @@ rspamadm_rescore (gint argc, gchar **argv) {
 				"logdir", 0, false);
 		ucl_object_insert_key (cfg->rcl_obj, ucl_object_fromstring (output),
 				"output", 0, false);
+		ucl_object_insert_key (cfg->rcl_obj, ucl_object_fromint (threads),
+				"threads", 0, false);
+		ucl_object_insert_key (cfg->rcl_obj, ucl_object_frombool (verbose),
+				"verbose", 0, false);
+		ucl_object_insert_key (cfg->rcl_obj, ucl_object_fromstring (iters),
+				"i", 0, false);
 		ucl_object_insert_key (cfg->rcl_obj, ucl_object_frombool (score_diff),
```
Comment (vstakhov) [inline src/rspamadm/rescore.c]: You don't need to do it, it is handled by Lua.

File: src/rspamadm/rescore.c [hunk 2]
```diff
@@ -48,11 +51,17 @@ static GOptionEntry entries[] = {
...
 				"Print score diff",                             NULL},
 		{"config", 'c', 0, G_OPTION_ARG_STRING, &config,
 				"Config file to use",     NULL},
+		{"threads", 't', 0, G_OPTION_ARG_INT, &threads,
+				"No of threads to use [Default:1]",     NULL},
+		{"verbose", 'v', 0, G_OPTION_ARG_NONE, &verbose,
+				"Enable verbose",     NULL},
+		{"iters", 'i', 0, G_OPTION_ARG_INT, &iters,
+				"Max iterations for perceptron [Default: 500]",     NULL},
 		{NULL,     0,   0, G_OPTION_ARG_NONE, NULL, NULL,       NULL}
```
Comment (vstakhov) [inline src/rspamadm/rescore.c]: Same here, please don't add any options to the C part.

File: src/rspamadm/rescore.c [hunk 3]
```diff
@@ -28,6 +28,9 @@ static gchar *logdir = NULL;
 static gchar *output = "new.scores";
 static gboolean score_diff = false;  /* Print score diff flag */
 static gchar *config = NULL;
+static gboolean verbose = true;
+static gint threads = 1;
+static gint iters = 500;
```
Comment (vstakhov) [inline src/rspamadm/rescore.c]: Not needed.

File: src/rspamadm/rescore.c [hunk 4]
```diff
@@ -83,10 +92,13 @@ rspamadm_rescore_help (gboolean full_help) {
...
-				"-o: scores output file location\n"
-				"-d: print scores diff\n"
-				"-i: max iterations for perceptron\n";
+				"-l: Logs of directory\n"
+				"-o: Scores output location\n"
+				"-d: Print score diff\n"
+				"-c: Config file to use\n"
+				"-t: No of threads to use\n"
+				"-v: Enable verbose\n"
+				"-i: Max iterations for perceptron [Default: 500]\n";
```
Comment (vstakhov) [inline src/rspamadm/rescore.c]: Default is `10` not `500`.

File: src/rspamadm/rescore.c [hunk 5]
```diff
@@ -83,10 +92,13 @@ rspamadm_rescore_help (gboolean full_help) {
...
-				"-l: path to logs directory\n"
-				"-o: scores output file location\n"
-				"-d: print scores diff\n"
-				"-i: max iterations for perceptron\n";
+				"-l: Logs of directory\n"
+				"-o: Scores output location\n"
+				"-d: Print score diff\n"
+				"-c: Config file to use\n"
+				"-t: No of threads to use\n"
+				"-v: Enable verbose\n"
```
Comment (vstakhov) [inline src/rspamadm/rescore.c]: It should be `--verbose`

File: src/rspamadm/rescore.c [hunk 6]
```diff
@@ -83,10 +92,13 @@ rspamadm_rescore_help (gboolean full_help) {
...
 				"Where options are:\n\n"
-				"-l: path to logs directory\n"
-				"-o: scores output file location\n"
-				"-d: print scores diff\n"
-				"-i: max iterations for perceptron\n";
+				"-l: Logs of directory\n"
+				"-o: Scores output location\n"
+				"-d: Print score diff\n"
+				"-c: Config file to use\n"
+				"-t: No of threads to use\n"
```
Comment (vstakhov) [inline src/rspamadm/rescore.c]: It should be `--threads`

**Comment by kakshay21** [comment]
@vstakhov Just changed the way you said. But all changes on the help command are lost.
screen shot 2018-03-04 at 7 52 35 pm

Seems like help function `rspamadm_rescore_help` is not activated with --help or -h command.

**Comment by kakshay21** [comment]
For reference, screenshot is based on this change
```
diff --git a/src/rspamadm/rescore.c b/src/rspamadm/rescore.c
index 4aa8a4e4c..a82bd095a 100644
--- a/src/rspamadm/rescore.c
+++ b/src/rspamadm/rescore.c
@@ -28,9 +28,6 @@ static gchar *logdir = NULL;
 static gchar *output = "new.scores";
 static gboolean score_diff = false;  /* Print score diff flag */
 static gchar *config = NULL;
-static gboolean verbose = true;
-static gint threads = 1;
-static gint iters = 500;
 extern struct rspamd_main *rspamd_main;
 /* Defined in modules.c */
 extern module_t *modules[];
@@ -56,12 +53,6 @@ static GOptionEntry entries[] = {
@@ -28,9 +28,6 @@ static gchar *logdir = NULL;
 static gchar *output = "new.scores";
 static gboolean score_diff = false;  /* Print score diff flag */
 static gchar *config = NULL;
-static gboolean verbose = true;
-static gint threads = 1;
-static gint iters = 500;
 extern struct rspamd_main *rspamd_main;
 extern struct rspamd_main *rspamd_main;
 /* Defined in modules.c */
 extern module_t *modules[];
@@ -56,12 +53,6 @@ static GOptionEntry entries[] = {
                                "Print score diff",                             NULL},
                {"config", 'c', 0, G_OPTION_ARG_STRING, &config,
                                "Config file to use",     NULL},
-               {"threads", 't', 0, G_OPTION_ARG_INT, &threads,
-                               "No of threads to use [Default:1]",     NULL},
-               {"verbose", 'v', 0, G_OPTION_ARG_NONE, &verbose,
-                               "Enable verbose",     NULL},
-               {"iters", 'i', 0, G_OPTION_ARG_INT, &iters,
-                               "Max iterations for perceptron [Default: 500]",     NULL},
                {NULL,     0,   0, G_OPTION_ARG_NONE, NULL, NULL,       NULL}
 };
 
@@ -96,9 +87,9 @@ rspamadm_rescore_help (gboolean full_help) {
                                "-o: Scores output location\n"
                                "-d: Print score diff\n"
                                "-c: Config file to use\n"
-                               "-t: No of threads to use\n"
-                               "-v: Enable verbose\n"
-                               "-i: Max iterations for perceptron [Default: 500]\n";
+                               "--threads: No of threads to use\n"
+                               "--verbose: Enable verbose\n"
+                               "-i: Max iterations for perceptron [Default: 10]\n";
        } else {
                help_str = "Estimate optimal symbol weights from log files";
        }
@@ -196,12 +187,6 @@ rspamadm_rescore (gint argc, gchar **argv) {
                                "logdir", 0, false);
                ucl_object_insert_key (cfg->rcl_obj, ucl_object_fromstring (output),
                                "output", 0, false);
-               ucl_object_insert_key (cfg->rcl_obj, ucl_object_fromint (threads),
-                               "threads", 0, false);
-               ucl_object_insert_key (cfg->rcl_obj, ucl_object_frombool (verbose),
-                               "verbose", 0, false);
-               ucl_object_insert_key (cfg->rcl_obj, ucl_object_fromstring (iters),
-                               "i", 0, false);
                ucl_object_insert_key (cfg->rcl_obj, ucl_object_frombool (score_diff),
                                "diff", 0, false);
                pcfg = lua_newuserdata (L, sizeof (struct rspamd_config *));
```

**Comment by kakshay21** [comment]
@vstakhov Not producing the expected result. I think we need to register those commands.

**Pull request closed without merging**